shopify-client/src/views/ShopList.vue

```diff
@@ -0,0 +1,261 @@
+        No results found
+      </div>
+      <div
+          v-else
+          v-for="shop in shops"
+          :key="shop.id"
+          class="shopTile"
+      >
+        <img :src="'/' + shop.image"/> <!--Update when root image location is known -->
+        <h3>
+          <a :href="'/shop/' + shop.id">
+            {{shop.name.toLocaleUpperCase()}}
+          </a>
+        </h3>
+        <p>
+          {{shop.description}}
+        </p>
+        <div
+            v-for="category in shop.categories"
+            :key="category.id"
+            :class="'shopAttribute shopCategory ' + category.name"
+        >
+          {{category.name}}
+        </div>
+        <div
+            v-for="tag in shop.tags"
+            :key="tag.id"
+            :class="'shopAttribute shopTag ' + tag.name"
+        >
+          {{tag.name}}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+//Placeholders
+//import header from '@/components/header.vue'
+//import footer from '@/components/footer.vue'
+
+let addDiff = (newAttributes, target) =>
+    newAttributes.forEach(function (attribute){
+      if(target.every(element => attribute.name !== element.name)){
+        target.push(attribute)
+      }
+    });
+
+//checker if the shop has the selected tag and categories
+let checker = (shop, target) =>
+    target.every(attribute => shop.getElementsByClassName(attribute).length > 0);
+
+export default {
+  name: "stopList",
+  data(){
+    return{
+      shops:[],
+      categories: [],
+      tags: [],
+      keyword: '',
+    }
+  },
+  created: function(){
+    const vm = this;
+    vm.keyword = new URLSearchParams(window.location.search).get('keyword');
+
+    if(vm.keyword === null){
+      vm.keyword = ''
+    }
+
+    fetch('/api/shops/shopList?keyword=' + vm.keyword)
+        .then((response) => response.json())
+        .then((response) => {
+          if (response.status == 404) {
+            throw new Error(response.message);
+          }
+          return response;
+        })
+        .then((data) => {
+          vm.shops = data;
+
+          data.forEach(function(shop){
+            addDiff(shop.categories, vm.categories);
+            addDiff(shop.tags, vm.tags);
+          });
+        })
+        .catch((error) => {
+          vm.error = error;
+        });
+  },
+  methods: {
+    filterSearch: function(event){
+      if(event.target.tagName == 'INPUT'){
+        let checkedBoxes = [];
+        let boxes = document.getElementsByClassName("checkBox");
+        let shops = document.getElementsByClassName("shopTile")
+
+        //Obtain checked boxes
+        boxes.forEach(function(box){
+          if(box.checked){
+            checkedBoxes.push(box.id) //box's id is the name of the tag/category
+          }
+        });
+
+        if(checkedBoxes.length > 0){
+          shops.forEach(function(shop){
+            if(checker(shop, checkedBoxes)){
+              shop.style.display = "inline-block";
```

Comment:
For filtering it's better to have a second list that holds all the shops loaded. For example, create another data property called `filteredShops`. Then you can just do `filteredShop = shops.filter(shop => checker(shop, checkedBoxes)). Your template would render the shops in `filteredShop` instead of `shops`. On load `filteredShops` = `shops`. This is a better way than toggling the display style.